### PR TITLE
Add some comments to scheduler.go and balancer.go

### PR DESCRIPTION
**Pull request by ming-relax**

### CODE REVIEW

**Comment by disksing** [comment]
Thanks @ming-relax . LGTM.

File: server/balancer.go [hunk 1]
```diff
@@ -636,6 +643,8 @@ func (h *balanceHotRegionScheduler) balanceByPeer(cluster *clusterInfo) (*Region
 	return nil, nil, nil
 }
 
+// selectDestStoreByPeer select a target store hold the region of the source region.
```
Comment (siddontang) [inline server/balancer.go]: selects a target

**Comment by siddontang** [comment]
PTAL @queenypingcap

**Comment by siddontang** [comment]
LGTM

File: server/balancer.go [hunk 2]
```diff
@@ -603,6 +610,8 @@ func (h *balanceHotRegionScheduler) balanceByPeer(cluster *clusterInfo) (*Region
 	return nil, nil, nil
 }
 
+// selectDestStoreByPeer selects a target store hold the region of the source region.
```
Comment (queenypingcap) [inline server/balancer.go]: -> selectDestStoreByPeer selects a target store to hold the region of the source region.

File: server/balancer.go [hunk 3]
```diff
@@ -603,6 +610,8 @@ func (h *balanceHotRegionScheduler) balanceByPeer(cluster *clusterInfo) (*Region
...
+// We choose a target store based on the hot region number of this store and written bytes of this store.
```
Comment (queenypingcap) [inline server/balancer.go]: -> We choose a target store based on the hot region number and written bytes of this store.

File: server/coordinator.go [hunk 4]
```diff
@@ -271,6 +271,8 @@ func (c *coordinator) addOperator(op Operator) bool {
 
 	log.Infof("[region %v] add operator: %+v", regionID, op)
 
+	// If the new operator passed in has higher priorities than old one,
```
Comment (queenypingcap) [inline server/coordinator.go]: than old one -> than the old one

File: server/scheduler.go [hunk 5]
```diff
@@ -275,6 +275,8 @@ func newRemovePeer(region *RegionInfo, peer *metapb.Peer) Operator {
 func newTransferPeer(region *RegionInfo, kind ResourceKind, oldPeer, newPeer *metapb.Peer) Operator {
 	addPeer := newAddPeerOperator(region.GetId(), newPeer)
 	removePeer := newRemovePeerOperator(region.GetId(), oldPeer)
+
+	// If the source peer is a leader, then we need to transfer leader
```
Comment (queenypingcap) [inline server/scheduler.go]: transfer leader -> transfer the leader to another region.

**Comment by siddontang** [comment]
Any update? @disksing @ming-relax

**Comment by siddontang** [comment]
Hi @ming-relax 

Please fix the conflict and I think we can go on now.

**Comment by ming-relax** [comment]
@siddontang fixed

**Comment by disksing** [comment]
PTAL @nolouch

**Comment by nolouch** [comment]
LGTM

**Pull request merged**